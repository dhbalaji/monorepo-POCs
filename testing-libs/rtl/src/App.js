import logo from "./logo.svg";
import "./App.css";

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React Again
        </a>
      </header>
      <div id="app">
        <label htmlFor="username-input">Username</label>
        <input id="username-input" />
      </div>
    </div>
  );
}

export default App;
