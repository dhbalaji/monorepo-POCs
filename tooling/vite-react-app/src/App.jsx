import { useState } from 'react'
import reactLogo from './assets/react.svg'
import './App.css'

function App({focusHandler}) {
  const [count, setCount] = useState(0)

  return (
    <div className="App">
      <div>
        <a href="https://vitejs.dev" target="_blank" rel="noreferrer">
          <img src="/vite.svg" className="logo" alt="Vite logo" />
        </a>
        <a href="https://reactjs.org" target="_blank" rel="noreferrer">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
        <a href="https://vitest.dev" target="_blank" rel="noreferrer">
          <img src="https://vitest.dev/logo-shadow.svg" className="logo react" alt="Vitest logo" />
        </a>
      </div>
      <h1>Vite + React + Vitest</h1>
      <div className="card">
        <button onClick={() => setCount((count) => count + 1)}>
          count is {count}
        </button>
        <button onFocus={() => {
          focusHandler?.();
        }
          } data-testid="doe">
          Dummy Btn
        </button>
        <p>
          Edit <code>src/App.jsx</code> and save to test HMR
        </p>
      </div>
      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p>
    </div>
  )
}

export default App
