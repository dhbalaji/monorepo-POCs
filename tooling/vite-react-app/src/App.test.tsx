import React from "react";
import jest from "jest-mock";
import App from "./App";
import { describe, it, expect } from "vitest";
import { findByRole, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("App test", () => {
  it("should contain text", async () => {
    render(<App />);
    // screen.debug();
    expect(screen.getByText("Vite + React + Vitest")).toBeDefined();
    expect(screen.queryAllByRole("link").length).toBe(3);
    expect(screen.queryAllByRole("link")[0].getAttribute("href")).toBe(
      "https://vitejs.dev"
    );
    expect(screen.queryAllByRole("link")[1].getAttribute("href")).toBe(
      "https://reactjs.org"
    );
    expect(screen.queryAllByRole("link")[2].getAttribute("href")).toBe(
      "https://vitest.dev"
    );

    const btn = screen.getByRole("button", {
      name: /count is 0/i,
    });
    expect(btn).toBeDefined();

    userEvent.click(btn);
    userEvent.click(btn);
    await screen.findByText(/count is 2/i);

    // This would fail because on click would change text in async way
    // expect(screen.getByRole("button").textContent).toBe();
  });

  it("should match snapshot", () => {
    const result = render(<App />);
    expect(result).toMatchSnapshot();
  });
});
