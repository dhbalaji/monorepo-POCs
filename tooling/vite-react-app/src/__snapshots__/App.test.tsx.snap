// Vitest Snapshot v1

exports[`App test > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div>
          <a
            href="https://vitejs.dev"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="Vite logo"
              class="logo"
              src="/vite.svg"
            />
          </a>
          <a
            href="https://reactjs.org"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="React logo"
              class="logo react"
              src="/src/assets/react.svg"
            />
          </a>
          <a
            href="https://vitest.dev"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="Vitest logo"
              class="logo react"
              src="https://vitest.dev/logo-shadow.svg"
            />
          </a>
        </div>
        <h1>
          Vite + React + Vitest
        </h1>
        <div
          class="card"
        >
          <button>
            count is 
            0
          </button>
          <button
            data-testid="doe"
          >
            Dummy Btn
          </button>
          <p>
            Edit 
            <code>
              src/App.jsx
            </code>
             and save to test HMR
          </p>
        </div>
        <p
          class="read-the-docs"
        >
          Click on the Vite and React logos to learn more
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div>
        <a
          href="https://vitejs.dev"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Vite logo"
            class="logo"
            src="/vite.svg"
          />
        </a>
        <a
          href="https://reactjs.org"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="React logo"
            class="logo react"
            src="/src/assets/react.svg"
          />
        </a>
        <a
          href="https://vitest.dev"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Vitest logo"
            class="logo react"
            src="https://vitest.dev/logo-shadow.svg"
          />
        </a>
      </div>
      <h1>
        Vite + React + Vitest
      </h1>
      <div
        class="card"
      >
        <button>
          count is 
          0
        </button>
        <button
          data-testid="doe"
        >
          Dummy Btn
        </button>
        <p>
          Edit 
          <code>
            src/App.jsx
          </code>
           and save to test HMR
        </p>
      </div>
      <p
        class="read-the-docs"
      >
        Click on the Vite and React logos to learn more
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
